### PR TITLE
Clean up the login logic

### PR DESCRIPTION
**Pull request by maffoo**
There was a subtle concurrency bug with the login logic which sometimes resulted in a login dialog popping up, even when the credentials had already been saved in local or session storage. This fixes that problem and also cleans up the logic overall to use promises and async functions more consistently.

### CODE REVIEW

File: client-js/app/scripts/app.ts [hunk 1]
```diff
@@ -636,7 +644,7 @@ window.addEventListener('WebComponentsReady', () => {
       throw new Error('no credentials');
     } else {
       try {
-        return mgr.login({
+        await mgr.login({
```
Comment (maffoo) [inline client-js/app/scripts/app.ts:651]: This was the bug; we were returning instead of waiting for the login to complete or fail.

**Comment by joshm-3b6b58** [comment]
Devil's advocate; if there are subtle concurrency bugs, should we have a unit test for this?

**Comment by maffoo** [comment]
In an ideal world, yes. But it's not really possible to unit test this as it's tied to polymer UI, etc.

**Comment by joshm-3b6b58** [comment]
LGTM

**Pull request merged**